geode-core/src/test/java/org/apache/geode/internal/cache/control/SerializableRestoreRedundancyResultsImplTest.java

```diff
@@ -153,39 +157,64 @@ public void addRegionResultsAddsToCorrectInternalMapAndAddsPrimaryReassignmentDe
+        .isEqualTo(underRedundancyRegionResult);
 
     Map<String, RegionRedundancyStatus> successfulRegionResults =
         results.getSatisfiedRedundancyRegionResults();
-    assertThat(successfulRegionResults.size(), is(1));
-    assertThat(successfulRegionResults.get(successfulRegionName), is(successfulRegionResult));
+    assertThat(successfulRegionResults.size()).isEqualTo(1);
+    assertThat(successfulRegionResults.get(successfulRegionName)).isEqualTo(successfulRegionResult);
 
-    assertThat(results.getTotalPrimaryTransfersCompleted(), is(transfersCompleted));
-    assertThat(results.getTotalPrimaryTransferTime().toMillis(), is(transferTime));
+    assertThat(results.getTotalPrimaryTransfersCompleted()).isEqualTo(transfersCompleted);
+    assertThat(results.getTotalPrimaryTransferTime()).isEqualTo(transferTime);
   }
 
   @Test
   public void addPrimaryDetailsUpdatesValue() {
-    assertThat(results.getTotalPrimaryTransfersCompleted(), is(0));
-    assertThat(results.getTotalPrimaryTransferTime().toMillis(), is(0L));
+    assertThat(results.getTotalPrimaryTransfersCompleted()).isEqualTo(0);
+    assertThat(results.getTotalPrimaryTransferTime()).isEqualTo(0L);
     results.addPrimaryReassignmentDetails(details);
-    assertThat(results.getTotalPrimaryTransfersCompleted(), is(transfersCompleted));
-    assertThat(results.getTotalPrimaryTransferTime().toMillis(), is(transferTime));
+    assertThat(results.getTotalPrimaryTransfersCompleted()).isEqualTo(transfersCompleted);
+    assertThat(results.getTotalPrimaryTransferTime()).isEqualTo(transferTime);
     results.addPrimaryReassignmentDetails(details);
-    assertThat(results.getTotalPrimaryTransfersCompleted(), is(transfersCompleted * 2));
-    assertThat(results.getTotalPrimaryTransferTime().toMillis(), is(transferTime * 2));
+    assertThat(results.getTotalPrimaryTransfersCompleted()).isEqualTo(transfersCompleted * 2);
+    assertThat(results.getTotalPrimaryTransferTime()).isEqualTo(transferTime * 2);
+  }
+
+  @Test
+  public void testSerializable() throws JsonProcessingException {
+
+    RestoreRedundancyResultsImpl restoreRedundancyResults = new RestoreRedundancyResultsImpl();
+    restoreRedundancyResults.setStatusMessage("Test");
+    restoreRedundancyResults.setSuccess(true);
+    restoreRedundancyResults.setTotalPrimaryTransfersCompleted(150);
+    restoreRedundancyResults.setTotalPrimaryTransferTime(250);
+    RegionRedundancyStatusImpl regionRedundancyStatus = new RegionRedundancyStatusImpl();
+    regionRedundancyStatus.setActualRedundancy(1);
+    regionRedundancyStatus.setConfiguredRedundancy(1);
+    regionRedundancyStatus.setRegionName("/foo");
+    regionRedundancyStatus.setStatus(SATISFIED);
+    restoreRedundancyResults.addRegionResult(regionRedundancyStatus);
+    String jsonString = geodeMapper.writeValueAsString(restoreRedundancyResults);
+    // deserialize the class
+    RestoreRedundancyResultsImpl value =
```

Comment:
you can use assertJ's `isEqualToComparingFieldByField` method to compare all the fields are the same

Reply:
done

Reply:
did you forget to push your commit?

Reply:
Nope it is failing... It doesn't like the objects in the satisfiedregion lists.

Reply:
sorry hash maps.

Reply:
assertThat(value).usingRecursiveComparison().isEqualTo(restoreRedundancyResults);

worked